### PR TITLE
[Objc] Added support for lower case discriminator

### PR DESCRIPTION
**Pull request by sylvain-abadie**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Add support for lowercase discriminator. Currently if the discriminator is "cat" (and not "Cat") the Cat class won't be deserialized. This just adds a test to deserialize by adding an uppercase to the discriminator.

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/objc/model-body.mustache [hunk 1]
```diff
@@ -28,6 +28,9 @@
          return [super initWithDictionary:dict error:err];
     }
     Class class = NSClassFromString([@"{{classPrefix}}" stringByAppendingString:discriminatedClassName]);
+    if(!class){
+      class = NSClassFromString([@"{{classPrefix}}" stringByAppendingString:[discriminatedClassName capitalizedString]]);
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/objc/model-body.mustache]: We prefer 4-space indentation. Please also add a space before `{` in line 31

**Comment by wing328** [comment]
Please also run `./bin/objc-petstore-all.sh` to update ObjC petstore sample.

**Comment by wing328** [comment]
Would this change cater both lowercase discriminator and discriminator in the original form (e.g. PascalCase) ?

**Comment by sylvain-abadie** [comment]
Yes the original behavior is preserved. In case the original behavior doesn't succeed, it tries to add an uppercase to the first letter of the discriminant.

**Comment by wing328** [comment]
@sylvain-abadie thanks for the explanation. PR merged into master.

**Pull request merged**